### ISSUE TITLE
associate-iam-oidc-provide when behind a proxy times out

### ISSUE THREAD

**Issue by boatmisser** [weaveworks/eksctl#1550]
**What happened?**
When behind a proxy running `associate-iam-oidc-provide` times out.

**What you expected to happen?**
`associate-iam-oidc-provide` to succeed

**How to reproduce it?**
Run `eksctl utils associate-iam-oidc-provider --cluster <cluster-name> --approve` from behind a proxy.

**Details**
When attempting to associate the OIC provider with my cluster from behind a proxy the request times out.
```
2019-11-14T08:32:02+13:00 [✖]  connecting to issuer OIDC (https://oidc.eks.ap-southeast-2.amazonaws.com:443/id/98A36B04EBD82D1D891245C85C520B78): dial tcp 13.238.186.19:443: connect: operation timed out
```
I can connect via curl to the URL and the IP that the message provides.
Also all other `eksctl` commands seem to work fine.

I had a look at the OIDC code and it appears to connect directly via `crypto/tls`; I am not familiar with that package, but I could not find any references to proxying the request; it does not appear to read the system proxy (`https_proxy`) nor provide the ability to pass one in.

I may be off base here, but is this an oversight?
Can anyone help with running `associate-iam-oidc-provide` from behind a proxy?

**Versions**
Please paste in the output of these commands:
```
$ eksctl version
[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.9.0"}

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T23:43:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.10-eks-5ac0f1", GitCommit:"5ac0f1d9ab2c254ea2b0ce3534fd72932094c6e1", GitTreeState:"clean", BuildDate:"2019-08-20T22:39:46Z", GoVersion:"go1.11.13", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by rubroboletus** [comment]
same here, please fix it, need to automate cluster creation in VPC including this functionality. Cannot open our intranet to Internet without proxy.

**Comment by wheestermans31** [comment]
Any update in this area? Same issue

**Comment by boatmisser** [comment]
Sadly no one replied to my issue; however it looks like they saw it as it as the issue has been fixed; see [#1619](https://github.com/weaveworks/eksctl/issues/1619).
The fix is in release 0.12.0 and I have tested it successfully in 0.13.0.

**Comment by wheestermans31** [comment]
True, looks fine with 0.13.0
